rust_src/src/interactive.rs

```diff
@@ -7,22 +7,76 @@ use crate::{
     remacs_sys::{EmacsInt, Qminus},
 };
 
+#[derive(Copy, Clone, Debug)]
+pub enum InteractiveNumericPrefix {
+    Object(LispObject),
+    Number(EmacsInt),
+}
+
+impl InteractiveNumericPrefix {
+    pub fn from_number(value: EmacsInt) -> Self {
+        Self::Number(value)
+    }
+
+    pub fn from_object(raw: LispObject) -> Self {
+        Self::Object(raw)
+    }
+
+    pub fn is_minus(self) -> bool {
+        match self {
+            Self::Number(_) => false,
+            Self::Object(raw) => raw.eq(Qminus),
+        }
+    }
+
+    pub fn unwrap(self) -> EmacsInt {
+        match self {
+            Self::Number(value) => value,
+            Self::Object(raw) => {
+                if raw.is_nil() {
+                    1
+                } else if raw.eq(Qminus) {
+                    -1
+                } else if raw.is_integer() {
+                    raw.as_fixnum_or_error()
```

Comment:
Didn't we drop this kind of methods ?

Reply:
Yes, we do usually. I left this one and the one in the next `else if` for clarity. Easy enough to switch them out for `into()` if that is the review consensus.